test_automation/work_order_utility.py

```diff
@@ -0,0 +1,171 @@
+    if work_order_id_json == "":
+        work_order_id = hex(random.randint(1, 2**64 -1))
+        input_json_temp["params"]["workOrderId"] = work_order_id
+    else:
+        work_order_id = work_order_id_json
+
+    # compute worker id
+    worker_id_json = input_json_temp["params"]["workerId"]
+    if worker_id_json == "":
+        input_json_temp["params"]["workerId"] = worker_obj.worker_id
+
+    # Convert workloadId to a hex string and update the request
+    workload_id = input_json_temp["params"]["workloadId"]
+    workload_id_hex = workload_id.encode("UTF-8").hex()
+    input_json_temp["params"]["workloadId"] = workload_id_hex
+    input_json_str1 = json.dumps(input_json_temp)
+
+    return input_json_str1, work_order_id
+
+def sign_work_order_request(input_json_str1, worker_obj, sig_obj, private_key):
+    """ Function to sign the work order request. """
+
+    logger.info("------------------Signing WorkOrderSubmit------------------")
+
+    # create session_iv through enclave_helper
+    session_iv = enclave_helper.generate_sessioniv()
+    # create encrypted_session_key from session_iv and worker_encryption_key
+    encrypted_session_key = enclave_helper.generate_encrypted_session_key(session_iv, worker_obj.worker_encryption_key)
+    # sign work order submit request
+    input_json_str1 = sig_obj.generate_client_signature(input_json_str1, worker_obj, private_key, session_iv, encrypted_session_key)
+
+    return input_json_str1, session_iv, encrypted_session_key
+
+def validate_work_order_response(response, check_submit):
+    """ Function to validate work order response. """
+
+    # check work order submit response
+    if "error" in check_submit.keys():
+        if response["error"]["code"] == check_submit["error"]["code"]:
+            err_cd = 0
+            logger.info("SUCCESS: WorkOrderSubmit - Request submitted and received expected error code.")
+        else:
+            err_cd = 1
+            logger.info("ERROR: Failed at WorkOrderSubmit - Request submitted and not received expected error code.")
+    elif "result" in check_submit.keys():
+        if response["result"]["code"] == check_submit["result"]["code"]:
+            err_cd = 0
+            logger.info("SUCCESS: WorkOrderSubmit - Request submitted and received expected error code")
+        else:
+            err_cd = 1
+            logger.info("ERROR: Failed at WorkOrderSubmit - Request submitted and not received expected result code.")
+    else:
+        logger.info("ERROR: Failed at WorkOrderSubmit - No Error or Result in response")
+        err_cd = 1
+
+    return err_cd
+
+def create_work_order_get_result(work_order_id):
+    """ Function to create work order get result request. """
+
+    logger.info("------------------Constructing WorkOrderGetResult------------------")
+    # create work order get result request
+    input_workorder_getresult = input_data = '{"jsonrpc": "2.0","method": "WorkOrderGetResult","id": 11,"params": {"workOrderId": ""}}'
```

Comment:
This line is too long and is hard to read. Please limit lines to 80 characters.

Reply:
Fixed line length to below 80